bindings.go

```diff
@@ -17,63 +17,141 @@ const (
 // declared and bound. If the ExchangeName is not defined (an empty string), the
 // queue will not be bound to the exchange but assumed to use the default match.
 type HandlerBinding struct {
-	QueueName    string
+	// QueueName is the name of the queue that the handler should be bound to.
+	QueueName string
+
+	// ExchangeName is the exchange that the queue should be bound to.
 	ExchangeName string
-	ExchangeType string
-	RoutingKey   string
-	BindHeaders  amqp.Table
-	Handler      HandlerFunc
+
+	// RoutingKey is the routing key that the queue should be bound to.
+	RoutingKey string
+
+	// BindHeaders is the headers that the queue should be bound to.
+	BindHeaders amqp.Table
+
+	// Handler is the function that should be called when a message is
+	// received.
+	Handler HandlerFunc
+
+	// QueueDurable sets the durable flag. Should not be used together with
+	// QueueExclusive since that will fail in a future RabbitMQ version.
+	QueueDurable bool
+
+	// Set the queue to be automatically deleted when the last consumer stops.
+	QueueAutoDelete bool
+
+	// SkipQueueDeclare will skip the queue declaration. You can be useful when
+	// you are migrating queue types or arguments and want to avoid the
+	// PRECONDITION_FAILED error. If this is set to true, you cannot use auto
+	// generated queue names.
+	SkipQueueDeclare bool
+
+	// Exclusive queues can only be used by the connection that created them.
+	// If this is true, the queue will be deleted during a network failure
+	// making any requests fail before the reconnect can happen.
+	QueueExclusive bool
+
+	// QueueArgs sets any extra queue arguments.
```

Comment:
Extra in addition to what? Can't find any defaults and setting `QueueArgs blindly will overwrite the default `x-queue-type` and x-expires` it seems like.

Reply:
Perhaps also a better name. `QueueDeclareArguments` perhaps? And I agree with your suggestion about the documentation.

Reply:
- [ ] Update docs
- [ ] Better name?

Reply:
I've added a `WithQueueDeclareArg(key, val)` function as well as a `WithQueueDeclareArgs(overwritingTable)` function.